library/include/Core.h

```diff
@@ -419,34 +433,60 @@ namespace DFHack
      *   The last step is to decrement Core::toolCount and wakeup main thread if
      *   no more tools are queued trying to acquire the
      *   Core::CoreSuspenderMutex.
+     *
+     * The public API for CoreSuspender only supports construction and destruction;
+     * all other locking operations are reserved
      */
-    class CoreSuspender : public CoreSuspenderBase {
+
+    class CoreSuspender : protected CoreSuspenderBase {
         using parent_t = CoreSuspenderBase;
     public:
-        CoreSuspender() : CoreSuspender{&Core::getInstance()} { }
-        CoreSuspender(std::defer_lock_t d) : CoreSuspender{&Core::getInstance(),d} { }
-        CoreSuspender(bool) : CoreSuspender{&Core::getInstance()} { }
-        CoreSuspender(Core* core, bool) : CoreSuspender{core} { }
-        CoreSuspender(Core* core) :
-            CoreSuspenderBase{core, std::defer_lock}
+        CoreSuspender() : CoreSuspender{Core::getInstance()} { }
+
+        CoreSuspender(Core& core) : CoreSuspenderBase{core}
         {
             lock();
         }
-        CoreSuspender(Core* core, std::defer_lock_t) :
-            CoreSuspenderBase{core, std::defer_lock}
-        {}
+
+        ~CoreSuspender() {
+            if (owns_lock()) unlock();
+        }
```

Comment:
are we avoiding making the destructor virtual for performance reasons? also, no need to check owns_lock(), neh?

Reply:
this is probably solely a result of the destructor being different before i refactored everything and i simply failed to notice that the implementation was the same

Reply:
actually looking at it more no, we need the explicit destructor. this is because the `CoreSuspenderBase` destructor will call `CoreSuspenderBase::unlock`, and we need the `CoreSuspender` destructor to call `CoreSuspender::unlock`

remember: none of these methods are virtual

Reply:
ah ok -- If would be worth adding a comment to highlight that for future readers